oteapi/strategies/resource/postgres.py

```diff
@@ -0,0 +1,176 @@
+"""Strategy class for application/vnd.postgresql"""
+# pylint: disable=unused-argument
+from typing import Any, Dict, Optional
+from urllib.parse import urlunparse
+
+import psycopg
+from pydantic import AnyUrl, Field, parse_obj_as, root_validator
+from pydantic.dataclasses import dataclass
+
+from oteapi.models import AttrDict, DataCacheConfig, ResourceConfig, SessionUpdate
+
+
+class PostgresConfig(AttrDict):
+    """Configuration data model for
+    [`PostgresResourceStrategy`][oteapi.strategies.resource.application_vnd_postgres.PostgresParseStrategy]."""
+
+    user: Optional[str] = Field(None, description="postgres server username")
+    dbname: Optional[str] = Field(None, description="postgres dbname name")
+    password: Optional[str] = Field(None, description="postgres password")
+
+    sqlquery: Optional[str] = Field("", description="A SQL query string.")
+
+
+class PostgresResourceConfig(ResourceConfig):
+    """Postgresql parse strategy config"""
+
+    configuration: PostgresConfig = Field(
+        PostgresConfig(),
+        description=(
+            "Configuration for resource. " "Values in the accessURL take precedence."
+        ),
+    )
+    datacache_config: Optional[DataCacheConfig] = Field(
+        None,
+        description="Configuration options for the local data cache.",
+    )
+
+    @classmethod
+    def _urlconstruct(
+        cls,  # PEP8 - Always use cls for the first argument to class methods.
+        scheme: Optional[str] = "",  # Schema defining link format
+        user: Optional[str] = None,  # Username
+        password: Optional[str] = None,  # Password
+        host: Optional[str] = None,
+        port: Optional[int] = None,
+        path: Optional[str] = "",
+        params: Optional[str] = "",
+        query: Optional[str] = "",
+        fragment: Optional[str] = "",
+    ):
+
+        """Construct a pydantic AnyUrl based on the given URL properties"""
+
+        # Hostname should always be given
+        if not host:
+            raise Exception("hostname must be specified")
```

Comment:
Use a more specific Exception subclass, like a subclass of ValueError called UrlError.

Reply:
Agreed with Jesper.

Reply:
Done